pkg/relay/ethereum/tx.go

```diff
@@ -29,41 +31,29 @@ func (c *Chain) SendMsgs(msgs []sdk.Msg) ([]core.MsgID, error) {
 		)
 		ctx := context.Background()
 		opts := c.TxOpts(ctx)
-		switch msg := msg.(type) {
-		case *clienttypes.MsgCreateClient:
-			tx, err = c.TxCreateClient(opts, msg)
-		case *clienttypes.MsgUpdateClient:
-			tx, err = c.TxUpdateClient(opts, msg)
-		case *conntypes.MsgConnectionOpenInit:
-			tx, err = c.TxConnectionOpenInit(opts, msg)
-		case *conntypes.MsgConnectionOpenTry:
-			tx, err = c.TxConnectionOpenTry(opts, msg)
-		case *conntypes.MsgConnectionOpenAck:
-			tx, err = c.TxConnectionOpenAck(opts, msg)
-		case *conntypes.MsgConnectionOpenConfirm:
-			tx, err = c.TxConnectionOpenConfirm(opts, msg)
-		case *chantypes.MsgChannelOpenInit:
-			tx, err = c.TxChannelOpenInit(opts, msg)
-		case *chantypes.MsgChannelOpenTry:
-			tx, err = c.TxChannelOpenTry(opts, msg)
-		case *chantypes.MsgChannelOpenAck:
-			tx, err = c.TxChannelOpenAck(opts, msg)
-		case *chantypes.MsgChannelOpenConfirm:
-			tx, err = c.TxChannelOpenConfirm(opts, msg)
-		case *chantypes.MsgRecvPacket:
-			tx, err = c.TxRecvPacket(opts, msg)
-		case *chantypes.MsgAcknowledgement:
-			tx, err = c.TxAcknowledgement(opts, msg)
-		// case *transfertypes.MsgTransfer:
-		// 	err = c.client.transfer(msg)
-		default:
-			logger.Error("failed to send msg", errors.New("illegal msg type"), "msg", msg)
-			panic("illegal msg type")
+		opts.GasLimit = math.MaxUint64
+		tx, err = c.SendTx(opts, msg, true)
+		if err != nil {
+			logger.Error("failed to send msg / NoSend: true", err, "msg", msg)
+			return nil, err
+		}
+		estimatedGas, err := c.EstimateGas(ctx, tx)
+		if err != nil {
+			logger.Error("failed to estimate gas", err, "msg", msg)
+			return nil, err
 		}
+
+		txGasLimit := uint64(float64(estimatedGas) * c.Config().GasEstimateMultiplier)
+		if txGasLimit > c.Config().MaxGasLimit {
+			return nil, fmt.Errorf("estimated gas %d exceeds max gas limit %d", txGasLimit, c.Config().MaxGasLimit)
+		}
```

Comment:
Instead of returning an error, we should adopt `Config().MaxGasLimit` as gas limit.
And, it's good to log WARN or INFO in such a case.

Reply:
Fixed